src/graphnet/data/pipeline.py

```diff
@@ -0,0 +1,173 @@
+            device (torch._device): the device used for computation
+            retro_table_name (str, optional): Name of the retro table for. Defaults to 'retro'.
+            outdir (path, optional): the directory in which the pipeline database will be stored. Defaults to None.
+            batch_size (int, optional): batch size for inference. Defaults to 100.
+            n_workers (int, optional): number of workers used in dataloading. Defaults to 10.
+            pipeline_name (str, optional): name of the pipeline. If such a pipeline already exists, an error will be prompted to avoid overwriting. Defaults to 'pipeline'.
+        """
+        self._pipeline_name = pipeline_name
+        self._device = device
+        self.n_workers =  n_workers
+        self._features = features
+        self._truth =  truth
+        self._batch_size = batch_size
+        self._outdir = outdir
+        self._module_dict = module_dict
+        self._retro_table_name = retro_table_name
+
+    def __call__(self, database, pulsemap) -> dict:
+        """Runs inference of each field in self._module_dict[target]['']
+
+        Args:
+            database (path): path to database with pulsemap and truth
+            pulsemap (str): name of pulsemaps
+        """
+        outdir = self._get_outdir(database)
+        if isinstance(self._device, str): # Because pytorch lightning insists on breaking pytorch cuda device naming scheme
+            device = int(self._device[-1])
+        if os.path.isdir(outdir) == False:
+            dataframes = []
+            for target in self._module_dict.keys():
+                dataloader = self._make_dataloader(database, pulsemap)
+                trainer = Trainer(gpus = [device])
+                model = torch.load(self._module_dict[target]['path'], map_location = 'cpu', pickle_module = dill)
+                model._gnn.to(self._device)
+                model._device = self._device
+                model.to(self._device)
+                model.eval()
+                model.inference()
+                results = get_predictions(trainer,model, dataloader, self._module_dict[target]['output_column_names'], ['event_no'])
+                dataframes.append(results.sort_values('event_no').reset_index(drop = True))
+            df = self._combine_outputs(dataframes)
+            self._make_pipeline_database(outdir,df, database)
+        else:
+            print(outdir)
+            print('WARNING - Pipeline named %s already exists! \n Please rename pipeline!'%self._pipeline_name)
+        return
+    def _load_modules(self, module_dict):
+        """Loads each module in _module_dict
+
+        Args:
+            module_dict (dict): dictionary of GNN ModuleList
+
+        Returns:
+            dict: module_dict with "loaded_module" field added.
+        """
+        for target in module_dict.keys():
+            model = torch.load(module_dict[target]['path'], map_location = 'cpu', pickle_module = dill)
```

Comment:
Hey Rasmus, I was recently doing some model loading and pytorch often suggests loading from the state_dict as sometimes loading the full model can result in funky errors (see https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_models_for_inference.html ) wondered if you had considered loading in from state_dict as we also save that afaik